dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/report/ValidationErrorReporter.java

```diff
@@ -54,42 +65,71 @@
 
     private final TrackerImportValidationContext validationContext;
 
-    private final Class<?> mainKlass;
-
-    private final AtomicInteger listIndex = new AtomicInteger( 0 );
-
+    /*
+     * The Tracker object uid to which this reporter is associated.
+     */
```

Comment:
We could maybe call this another way: at least "mainUid".
I personally do not like how the word main is used here because it makes you think that there are other less important uids somewhere.

Maybe just uid would make the job.

Reply:
Good thinking Enrico,
mainId indicates there is another lesser id. Alternatively we can do primaryId or just id/uid as you said.

I will merge despite this though, I think we can clean up that later